txnkv/transaction/pessimistic.go

```diff
@@ -182,10 +183,23 @@ func (action actionPessimisticLock) handleSingleBatch(c *twoPhaseCommitter, bo *
 				c.run(c, action.LockCtx)
 			}
 
-			if action.ReturnValues {
+			// Handle the case that the TiKV's version is too old and doesn't support `CheckExistence`.
+			// If `CheckExistence` is set, `ReturnValues` is not set and `CheckExistence` is not supported, skip
+			// retrieving value totally (indicated by `skipRetrievingValue`) to avoid panicking.
+			skipRetrievingValue := !action.ReturnValues && action.CheckExistence && len(lockResp.NotFounds) == 0
```

Comment:
Could the `len(lockResp.NotFounds) == 0` represent such situation accurately?

Reply:
The previous version of TiKV already supports setting the `NotFounds` field when `ReturnValues` is set. So when `ReturnValues` is set, both new and old versions of TiKV should work. But if `CheckExistence` is set and `ReturnValues` is unset, since old TiKV cannot recognize the new field `CheckExistence`, the `NotFounds` returned from TiKV will be empty. We cannot retrieve existence in this case, and it should be the only case.